dwave/plugins/torch/boltzmann_machine.py

```diff
@@ -210,35 +364,216 @@ def interactions(self, x: torch.Tensor) -> torch.Tensor:
+                biases with.
+            quadratic_range (tuple[float, float], Optional): The minimum and maximum values to clip
+                quadratic biases with.
+
+        Returns:
+            tuple[dict, dict]: The linear and quadratic biases in dictionary format compatible with
+            `dimod.Sampler.sample_ising`.
+        """
+        linear = prefactor * self._linear.detach()
+        quadratic = prefactor * self._quadratic.detach()
+        if linear_range is not None:
+            linear = linear.clip(*linear_range)
+        if quadratic_range is not None:
+            quadratic = quadratic.clip(*quadratic_range)
 
-    @property
-    def _ising(self) -> tuple[dict, dict]:
-        """Convert the model to Ising format"""
-        linear_biases = self.h.detach().cpu().tolist()
         edge_idx_i = self.edge_idx_i.detach().cpu().tolist()
         edge_idx_j = self.edge_idx_j.detach().cpu().tolist()
-        quadratic_bias_list = self.J.detach().cpu().tolist()
-        quadratic_biases = {
-            (a, b): w for a, b, w in zip(edge_idx_i, edge_idx_j, quadratic_bias_list)
+        h = {self._idx_to_node[i]: b for i, b in enumerate(linear.cpu().tolist())}
+        J = {
+            (self._idx_to_node[a], self._idx_to_node[b]): w
+            for a, b, w in zip(edge_idx_i, edge_idx_j, quadratic.cpu().tolist())
         }
+        return h, J
+
+    def _pad(self, x: torch.Tensor) -> torch.Tensor:
+        """Pads the observed spins with ``torch.nan``s at ``self.hidx`` to mark them as
+        hidden units.
+
+        Args:
+            x (torch.Tensor): Partially-observed spins of shape (b, N) where b is the
+                batch size and N is the number of visible units in the model.
+
+        Raises:
+            ValueError: Fully-visible models should not ``_pad`` data.
+
+        Returns:
+            torch.Tensor: A (b, N) tensor of spin variables where N is the total number
+            of variables, i.e., number of visible and hidden units.
+        """
+        bs = x.shape[0]
+        padded = torch.nan * torch.ones((bs, self._n_nodes), device=x.device)
+        padded[:, self.visible_idx] = x
+        return padded
+
+    def estimate_beta(self, spins: torch.Tensor) -> float:
+        """Estimate the maximum pseudolikelihood temperature using
+        ``dwave.system.temperatures.maximum_pseudolikelihood_temperature``.
+
+        Args:
+            spins (torch.Tensor): A tensor of shape (b, N) where b is the sample size,
+                and N denotes the number of variables in the model.
+
+        Returns:
+            float: The estimated inverse temperature of the model.
+        """
+        bqm = BinaryQuadraticModel.from_ising(*self.to_ising(1))
+        beta = 1 / mple(bqm, (spins.detach().cpu().numpy(), self._nodes))[0]
+        return beta
+
+    def _compute_effective_field(self, padded: torch.Tensor) -> torch.Tensor:
```

Comment:
Maybe move up and place `_compute_effective_field`, `_approximate_expectation_sampling`, `_compute_expectation_disconnected`, and `_pad` after `quasi_objective()` since they're only used there (I believe).